app/components/curriculum-inventory-sequence-block-overview.js

```diff
@@ -121,255 +127,206 @@ export default Component.extend({
-    const report = yield sequenceBlock.get('report');
-    const parent = yield sequenceBlock.get('parent');
-    let academicLevels = yield report.get('academicLevels');
-    academicLevels = academicLevels.toArray();
-    let isInOrderedSequence = false;
-    const orderInSequenceOptions = [];
-    if (isPresent(parent) && parent.get('isOrdered')) {
-      isInOrderedSequence = true;
-      const siblings = yield parent.get('children');
-      for (let i = 0, n = (siblings.toArray().length); i < n; i++) {
-        const num = i + 1;
-        orderInSequenceOptions.push(num);
-      }
+  }
+
+  /**
+   * Returns a list of published sessions that belong to a given course.
+   */
+  async getSessions(course) {
+    if (! course) {
+      return [];
     }
-    const linkedSessions = yield sequenceBlock.get('sessions');
-    const academicLevel = yield sequenceBlock.get('academicLevel');
-    const required = '' + sequenceBlock.get('required');
-    const duration = sequenceBlock.get('duration');
-    const startDate = sequenceBlock.get('startDate');
-    const endDate = sequenceBlock.get('endDate');
-    const childSequenceOrder = '' + sequenceBlock.get('childSequenceOrder');
-    const orderInSequence = sequenceBlock.get('orderInSequence');
-    const description = sequenceBlock.get('description');
-    const course = yield sequenceBlock.get('course');
-    const minimum = sequenceBlock.get('minimum');
-    const maximum = sequenceBlock.get('maximum');
-    this.setProperties({
-      parent,
-      report,
-      academicLevel,
-      academicLevels,
-      isInOrderedSequence,
-      orderInSequenceOptions,
-      startDate,
-      endDate,
-      duration,
-      childSequenceOrder,
-      orderInSequence,
-      description,
-      linkedSessions,
-      course,
-      required,
-      minimum,
-      maximum,
+    const sessions = await this.store.query('session', {
+      filters: {
+        course: course.get('id'),
+        published: true
+      },
     });
-  }),
-
-  saveCourseChange: task(function* (course) {
-    const block = this.sequenceBlock;
-    const oldCourse = block.get('course');
-    if (oldCourse !== course) {
-      block.set('sessions', []);
-      block.set('excludedSessions', []);
+    return sessions.toArray();
+  }
+
+  @action
+  changeRequired() {
+    this.args.sequenceBlock.set('required', parseInt(this.required, 10));
+    if ('2' === this.required) {
+      this.args.sequenceBlock.set('minimum', 0);
+    }
+    this.args.sequenceBlock.save();
+  }
+
+  @action
+  setRequired(required) {
+    this.required = required;
+    if ('2' === required) {
+      this.minimum = 0;
+    } else {
+      this.minimum = this.args.sequenceBlock.get('minimum');
+    }
+  }
+
+  @action
+  revertRequiredChanges() {
+    this.required = this.args.sequenceBlock.get('required').toString();
+    this.minimum = this.args.sequenceBlock.get('minimum');
+  }
+
+  @action
+  async saveCourse() {
```

Comment:
Async actions should always be tasks to avoid trying to call `this.args.sequenceBlock.set('sessions', []);` if the component is removed from the DOM between the promise starting and resolving.

Reply:
i updated all async actions to droppable task, this works out just fine. 

however. when saving linked courses in the ui, the console now shows a new warning. 

> ember-concurrency detected a potentially hazardous "self-cancel loop" between parent task `saveData` and child task `saveCourse`. If you want child task `saveCourse` to be canceled when parent task `saveData` is canceled, please change `.perform()` to `.linked().perform()`. If you want child task `saveCourse` to keep running after parent task `saveData` is canceled, change it to `.unlinked().perform()`

this _only_ happens on `saveCourse()`, the other edit-in-place controls do not throw this warning. seems to be working without problems, but may be something that needs to be handled in the `<EditableField> component, which lives in `common`.

Reply:
I vaguely recall seeing this before and not fixing it create https://github.com/ilios/common/issues/1577 so we're tracking it this time.